libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -2010,9 +2011,33 @@ void ExpressionCompiler::appendExternalFunctionCall(
 
 	utils().popStackSlots(remainsSize);
 
-	if (returnSuccessCondition)
+	if (returnSuccessConditionAndReturndata)
 	{
-		// already there
+		// success condition is already there
+		// The return parameter types can be empty, when this function is used as
+		// an internal helper function e.g. for ``send`` and ``transfer``. In that
+		// case we're only interested in the success condition, not the return data.
+		if (!_functionType.returnParameterTypes().empty())
+		{
+			if (haveReturndatacopy)
+			{
+				solAssert(haveReturndatacopy, "");
+				m_context << Instruction::RETURNDATASIZE;
+				m_context.appendInlineAssembly(R"({
+                    switch v case 0 {
```

Comment:
Yeah - I wasn't worried about calling ``returndatasize()`` multiple times, but just wanted something convenient on the stack that can be used both as input and output for the assembly  - but I should probably have more trust in the optimizer rather than trying to save a few instructions here and there like this...